src/core/DY.Core.Trace.Type.fst

```diff
@@ -48,7 +52,7 @@ type trace_event_ (label_t:Type) =
   // A random number has been generated, with some usage and label.
   | RandGen: usg:usage -> label_t -> len:nat{len <> 0} -> trace_event_ label_t
   // A state of a principal has been corrupt.
-  | Corrupt: prin:principal -> sess_id:state_id -> trace_event_ label_t
+  | Corrupt: time:timestamp -> trace_event_ label_t
```

Comment:
This is to point to a particular `SetState` event in the trace, rather than using `principal` and `state_id` to refer to a all the `SetState` events corresponding to this `principal` and `state_id`.

It's not that weird to use in practice, see for example the lemma stating that corrupted states are publishable:
https://github.com/REPROSEC/dolev-yao-star-extrinsic/blob/741561406bc3aa2644274c4193c60ccb5c234e63/src/core/DY.Core.Attacker.Knowledge.fst#L194-L203

Reply:
I see, since we currently don't have version numbers, the principal and session ID are not together enough to pick out a particular StateSet event. If we do have (unique) version numbers, for instance as a result of the proposed state changes (or a modified version thereof) in #58, I believe the tuple (principal, session ID, version ID) is sufficient to pick out a single `SetState` event. Of course this is in some sense interchangeable with the timestamp (I believe either can be used to look up the other), so I suppose the main question is which is better to use from a practical point of view --- timestamps are certainly simpler in that they're a single piece of data, but it's also a piece of data that we've largely tried to hide from the user of DY* when possible.

Reply:
Indeed, if at some point we want to change the `Corrupt` event to something else (e.g. using versions), this should be an easy change anyway.